### PR TITLE
Don't show cancel option once a user has linked paypal details

### PR DESCRIPTION
**Pull request by johnbotris**
had to do add some laziness to the dialog since the result of showActionDialog is somewhat self contained

### CODE REVIEW

File: src/api/common/utils/Utils.js [hunk 1]
```diff
@@ -400,3 +400,14 @@ export function typedKeys<K: string, V>(obj: {[K]: V}): Array<K> {
 	return downcast(Object.keys(obj))
 }
 
+export function getAsNotLazy<T>(maybe: MaybeLazy<T>): T {
```
Comment (charlag) [inline src/api/common/utils/Utils.js:403]: I don't like the name with "not", maybe "unwrapMaybeLazy" or "getFromLazy"?

**Review by charlag — COMMENTED** [review]
Rebase please!

File: src/gui/base/Dialog.js [hunk 2]
```diff
@@ -512,8 +513,8 @@ export class Dialog {
 		}
 
 		const actionBarAttrs: DialogHeaderBarAttrs = {
-			left: allowCancel ? [{label: cancelActionTextId, click: doCancel, type: ButtonType.Secondary}] : [],
-			right: okAction ? [{label: okActionTextId, click: doAction, type: ButtonType.Primary}] : [],
+			left: mapLazily(allowCancel, allow => allow ? [{label: cancelActionTextId, click: doCancel, type: ButtonType.Secondary}] : []),
+			right: okAction ? [{label: mapLazily(okActionTextId, id => lang.get(id)), click: doAction, type: ButtonType.Primary}] : [],
```
[the remark this answers is not in the document]
Comment (charlag) [reply src/gui/base/Dialog.js:517]: I forgot it's another type, I wouldn't complicate MaybeLazy with the second type tho, were wer always call lang.get and I didn't see it, just leave it as-is

File: src/api/common/utils/Utils.js (hunk 1 above)
Comment (johnbotris) [reply src/api/common/utils/Utils.js:403]: im gonna go with "resolveMaybeLazy"

Comment (charlag) [reply src/api/common/utils/Utils.js:403]: Sounds better than unwrap, yeah

File: src/subscription/PaymentMethodInput.js [hunk 3]
```diff
@@ -199,6 +202,6 @@ class PaypalInput {
 	}
 }
 
-function isPaypalAssigned(accountingInfo: AccountingInfo) {
+export function isPaypalAssigned(accountingInfo: AccountingInfo): boolean {
```
Comment (charlag) [inline src/subscription/PaymentMethodInput.js:205]: It's not used outside, why it's exported?

File: src/gui/base/Dialog.js [hunk 4]
```diff
@@ -524,14 +526,12 @@ export class Dialog {
 			]
 		}).setCloseHandler(doCancel)
 
-		if (allowCancel) {
```
Comment (charlag) [inline src/gui/base/Dialog.js:527]: If you always add this shortcut then you should add `enabled` to it

File: src/subscription/PaymentMethodInput.js (hunk 3 above)
Comment (johnbotris) [reply src/subscription/PaymentMethodInput.js:205]: oh yeah i changed it to use outside but then i unused it and didn't unchange

**Review by charlag — APPROVED** [review]

**Pull request merged**